analyzers/src/SonarAnalyzer.Common/Extensions/ITypeSymbolExtensions.cs

```diff
@@ -112,20 +106,6 @@ public static bool IsAny(this ITypeSymbol typeSymbol, ImmutableArray<KnownType>
     public static bool IsNullableOf(this ITypeSymbol type, KnownType typeArgument) =>
         NullableTypeArgument(type).Is(typeArgument);
 
-    public static bool IsType(this IParameterSymbol parameter, KnownType type) =>
-        parameter != null && parameter.Type.Is(type);
-
-    public static bool IsInType(this ISymbol symbol, KnownType type) =>
-        symbol != null && symbol.ContainingType.Is(type);
-
-    public static bool IsInType(this ISymbol symbol, ITypeSymbol type) =>
-        symbol?.ContainingType != null && symbol.ContainingType.Equals(type);
-
-    public static bool IsInType(this ISymbol symbol, ImmutableArray<KnownType> types) =>
-        symbol != null && symbol.ContainingType.IsAny(types);
```

Comment:
New styling rule idea: Methods in an extension class should extend the same type.

Reply:
#9227